apps/aecore/src/aec_upnp.erl

```diff
@@ -66,18 +88,27 @@ handle_cast(Other, State) ->
     epoch_sync:warning("Received unknown cast: ~p", [Other]),
     {noreply, State}.
 
-handle_info(add_port_mapping, State) ->
-    proc_lib:spawn(fun() -> add_port_mapping() end),
+handle_info(add_port_mapping, State0) ->
+    ok = kill_mapping_worker(State0),
+    State = spawn_mapping_worker(),
+
     %% Give additional 10 secs for UPnP/NAT-PMP discovery and setup, to
     %% make sure there is continuity in port mapping.
     erlang:send_after(1000 * (?MAPPING_LIFETIME - 10), self(), add_port_mapping),
     {noreply, State};
+handle_info({'EXIT', _Pid, normal}, State) ->
+    %% Received when calling delete_port_mapping/0 during prep_stop
```

Comment:
So there are some spawn with link inside the `nat:*` calls in `delete_port_mapping`. (Did not check.) Ok.